packages/common/lib/dpop/DPoP.ts

```diff
@@ -0,0 +1,242 @@
+) {
+  // There is not more than one DPoP HTTP request header field.
+  const dpop = request.headers['dpop'];
+  if (!dpop || typeof dpop !== 'string') {
+    throw new Error('missing or invalid dpop header. Expected compact JWT');
+  }
+
+  // The DPoP HTTP request header field value is a single and well-formed JWT.
+  const { header: dPoPHeader, payload: dPoPPayload } = parseJWT<JwtHeader, JwtPayload & Partial<DPoPJwtPayloadProps>>(dpop);
+
+  // Ensure all required header claims are present
+  if (dPoPHeader.typ !== 'dpop+jwt' || !dPoPHeader.alg || !dPoPHeader.jwk || typeof dPoPHeader.jwk !== 'object' || dPoPHeader.jwk.d) {
+    throw new Error('invalid_dpop_proof. Invalid header claims');
+  }
+
+  // Ensure all required payload claims are present
+  if (!dPoPPayload.htm || !dPoPPayload.htu || !dPoPPayload.iat || !dPoPPayload.jti) {
+    throw new Error('invalid_dpop_proof. Missing required claims');
+  }
+
+  // Validate alg is supported
+  if (options?.acceptedAlgorithms && !options.acceptedAlgorithms.includes(dPoPHeader.alg)) {
+    throw new Error(`invalid_dpop_proof. Invalid 'alg' claim '${dPoPHeader.alg}'. Only ${options.acceptedAlgorithms.join(', ')} are supported.`);
+  }
+
+  // Validate nonce if provided
+  if ((options?.expectedNonce && !dPoPPayload.nonce) || dPoPPayload.nonce !== options.expectedNonce) {
+    throw new Error('invalid_dpop_proof. Nonce mismatch');
+  }
+
+  // Verify JWT signature
+  try {
+    const verificationResult = await options.jwtVerifyCallback(
+      {
+        method: 'jwk',
+        type: 'dpop',
+        jwk: dPoPHeader.jwk,
+        alg: dPoPHeader.alg,
+      },
+      {
+        header: dPoPHeader,
+        payload: dPoPPayload,
+        raw: dpop,
+      },
+    );
+
+    if (!verificationResult) {
+      throw new Error('invalid_dpop_proof. Invalid JWT signature');
+    }
+  } catch (error: unknown) {
+    throw new Error('invalid_dpop_proof. Invalid JWT signature. ' + (error instanceof Error ? error.message : 'Unknown error'));
+  }
+
+  // Validate htm claim
+  if (dPoPPayload.htm !== request.method) {
+    throw new Error(`invalid_dpop_proof. Invalid htm claim. Must match request method '${request.method}'`);
+  }
+
+  // The htu claim matches the HTTP URI value for the HTTP request in which the JWT was received, ignoring any query and fragment parts.
+  const currentUri = request.fullUrl.split('?')[0].split('#')[0];
+  if (dPoPPayload.htu !== currentUri) {
+    throw new Error('invalid_dpop_proof. Invalid htu claim');
+  }
+
+  // Validate nonce if provided
+  if ((options.expectedNonce && dPoPPayload.nonce !== options.expectedNonce) || (!options.expectedNonce && dPoPPayload.nonce)) {
+    throw new Error('invalid_dpop_proof. Nonce mismatch');
+  }
+
+  // Validate iat claim
+  const { nowSkewedPast, nowSkewedFuture } = getNowSkewed(options.now);
+  if (
+    // iat claim is to far in the future
+    nowSkewedPast - (options.maxIatAgeInSeconds ?? 300) > dPoPPayload.iat ||
```

Comment:
I would make the defaults a bit smaller. Systems being of 5 minutes would be very bad. Suggest to use 60